src/instructlab/model/llmaaj.py

```diff
@@ -0,0 +1,164 @@
+    JSONL: str = "jsonl"
+    XLSX: str = "xlsx"
+    QNA_YAML: str = "qna.yaml"
+
+def model_is_local(model: str) -> bool:
+    if os.path.exists(model):
+        model_path = pathlib.Path(model)
+        valid_model = False
+        if model_path.is_dir():
+            valid_model = is_model_safetensors(model_path)
+        elif model_path.is_file():
+            valid_model = is_model_gguf(model_path)
+        return valid_model
+    else:
+        return False
+
+def get_endpoint_model_name(endpoint: str, api_key: str, http_client: httpx.Client) -> str:
+    try:
+        client = OpenAI(
+            base_url=endpoint,
+            api_key=api_key,
+            timeout=DEFAULTS.CONNECTION_TIMEOUT,
+            http_client=http_client,
+        )
+        models = client.models.list()
+    except OpenAIError as exc:
+        raise client_utils.ClientException(f"Connection Error {exc}") from exc
+
+    if len(models.data) != 1:
+        raise client_utils.ClientException(f"More than one model at endpoint, pass in model name manually")
+
+    return models.data[0].id
+
+def create_results_file_name(file_format: IOFileType, output_dir: str) -> str:
+    file_type = "jsonl"
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    if IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.info(f"Output dir provided is not a directory or does not exist, writing results to output directory {output_dir}")
+
+    return f"{output_dir}/responses-{timestamp}.{file_type}"
+
+def run_llmaaj(ctx, model, max_workers, gpus, backend, enable_serving_output, input_questions, output_file_formats, output_dir, model_prompt, temperature, model_name, judge_model_name):
+    logger.info(f"Input questions path is {input_questions}")
+
+    try:
+        student_api_key = None
+        if model_is_local(model):
+            logger.info(f"Model is local")
+            if model_name == None:
+                model_name = get_model_name(model)
+            server, api_base, effective_gpus = launch_server(
+                eval_serve=ctx.obj.config.serve,
+                tls_client_cert=ctx.params["tls_client_cert"],
+                tls_client_key=ctx.params["tls_client_key"],
+                tls_client_passwd=ctx.params["tls_client_passwd"],
+                tls_insecure=ctx.params["tls_insecure"],
+                model=model,
+                model_name=model_name,
+                max_workers=max_workers,
+                gpus=gpus,
+                backend=backend,
+                enable_serving_output=enable_serving_output,
+            )
+        else:
+            logger.info(f"Model is an endpoint")
+            student_api_key = os.environ.get("STUDENT_ENDPOINT_API_KEY", None)
```

Comment:
Does MT-Bench use this value too? If not, I recommend dropping it. Otherwise it might create inconsistent behavior.